File: Documents/CodeGym/Danny Store/FE/danny_store/src/service/ProductService.js
import axios from "axios";

const URL = "http://localhost:8080/api/admin/home/";
export const getProducts = async (type_id, category_id, concept_id, size_id, color_id, sort_name) => {
    try {
        const res = await axios.get(URL + `get-products?type_id=${type_id}&category_id=${category_id}&concept_id=${concept_id}&size_id=${size_id}&color_id=${color_id}&sort_name=${sort_name}`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getLatestProducts = async () => {
    try {
        const res = await axios.get(URL + 'get-latest-products');
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getProductsByConceptId = async (concept_id, limit) => {
    try {
        const res = await axios.get(URL + `get-products?concept_id=${concept_id}&limit=${limit}`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getSizesByNameAndColorId = async (name, colorId) => {
    try {
        const res = await axios.get(URL + `get-sizes-by-name-and-color-id?name=${name}&color_id=${colorId}`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getBestsellers = async () => {
    try {
        const res = await axios.get(URL + 'get-bestsellers');
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getColorsByName = async (name) => {
    try {
        const res = await axios.get(URL + `get-colors-by-name?name=${name}`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getColors = async () => {
    try {
        const res = await axios.get(URL + `get-colors-by-name`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getSizes = async () => {
    try {
        const res = await axios.get(URL + `get-sizes`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getProductByNameAndColorId = async (name, color_id) => {
    try {
        const res = await axios.get(URL + `get-product-by-name-and-color-id?name=${name}&color_id=${color_id}`);
        return res.data
    } catch (error) {
        console.log(error);
    }
}


export const getCategoriesByTypeId = async (typeId) => {
    try {
        const res = await axios.get(URL + `get-categories-by-type-id?type_id=${typeId}`);
        return res.data
    } catch (error) {
        console.log(error);
    }
}

export const getProductByIdAndSizeId = async (product_id, size_id) => {
    try {
        const res = await axios.get(URL + `get-product-by-id-and-size-id?product_id=${product_id}&size_id=${size_id}`)
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const checkSoldOut = async (product_id, size_id) => {
    try {
        const res = await axios.get(URL + `check-sold-out?product_id=${product_id}&size_id=${size_id}`)
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getProductById = async (id) => {
    try {
        const res = await axios.get(URL + `get-product-by-id?id=${id}`);
        return res.data;
    } catch (error) {
        console.log(error);
    }
}

export const getImagesByProductId = async (id) => {
    try {
        console.log(id);
        const res = await axios.get(URL + `get-images-by-product-id?product_id=${id}`);

        return res.data;
    } catch (error) {
        console.log(error);
    }
}

File: Documents/CodeGym/Danny Store/BE/danny_store/src/main/java/com/danny_store/service/product/ProductService.java
package com.danny_store.service.product;

import com.danny_store.dto.product.*;
import com.danny_store.repository.IProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository productRepository;

    @Override
    public List<IProductDto> getProducts(String type_id, String category_id, String concept_id, String size_id, String color_id,String sort_name,Integer limit) {
        return productRepository.getProducts(type_id, category_id, concept_id, size_id, color_id, sort_name, limit);
    }

    @Override
    public List<IColorDto> getColorsByName(String name) {
        return productRepository.getColorsByName(name);
    }

    @Override
    public List<ISizeDto> getSizesByNameAndColorId(String name, Integer color_id) {
        return productRepository.getSizesByNameAndColorId(name, color_id);
    }

    @Override
    public IProductDto getProductByNameAndColorId(String name, Integer color_id) {
        return productRepository.getProductByNameAndColorId(name, color_id);
    }

    @Override
    public List<IChildrenTypeDto> getCategoriesByTypeId(Integer type_id) {
        return productRepository.getCategoriesByTypeId(type_id);
    }

    @Override
    public List<IProductDto> getLatestProducts() {
        return productRepository.getLatestProducts();
    }

    @Override
    public List<ISizeDto> getSizes() {
        return productRepository.getSizes();
    }

    @Override
    public List<IProductDto> getBestsellers() {
        return productRepository.getBestsellers();
    }

    @Override
    public IProductDto getProductByIdAndSizeId(Integer product_id, Integer size_id) {
        return productRepository.getProductByIdAndSizeId(product_id, size_id);
    }

    @Override
    public Integer getQuantityByProductIdAndSizeId(Integer product_id, Integer size_id) {
        return productRepository.getQuantityByProductIdAndSizeId(product_id,size_id);
    }

    @Override
    public IProductDto getProductById(Integer product_id) {
        return productRepository.getProductsById(product_id);
    }

    @Override
    public List<IImageDto> getImagesByProductId(Integer product_id) {
        return productRepository.getImagesByProductId(product_id);
    }


}
